File: ur10_gripper_control/src/rosservice_client_example.cpp
#include "ros/ros.h"
#include <ur10_gripper_control/MoveRobot.h>
#include <geometry_msgs/Pose.h>
 
#include <cstdlib>
 
int main(int argc, char **argv) {
 
    // Initialize the node and set the name
    ros::init(argc, argv, "ur10_gripper_controller_client");
    
    ros::NodeHandle n;
    
    ros::ServiceClient client = n.serviceClient<ur10_gripper_control::MoveRobot>("ur10_gripper_controller/MoveRobot");
    
    // Create an instance of the srv request type
    ur10_gripper_control::MoveRobot srv;

    geometry_msgs::Pose pose;
    
    // Fill in the two values that will be sent to the Service Server
    srv.request.Pose = pose;
    
    // Call the service, and send the data
    if (client.call(srv))
    {
        ROS_INFO("Result: %d", (bool)srv.response.result);
    }
    else
    {
        ROS_ERROR("Failed to call service adder_server");
        return 1;
    }
    
    return 0;
}

File: ur10_gripper_control/scripts/rosservice_client_example.py
#!/usr/bin/env python

import rospy
from ur10_gripper_control.srv import MoveRobot
from geometry_msgs.msg import Pose

def move_robot_client():
    rospy.init_node('ur10_gripper_controller_client')

    # Wait for the service to become available
    rospy.wait_for_service('ur10_gripper_controller/MoveRobot')
    
    try:
        # Create a service proxy
        move_robot = rospy.ServiceProxy('ur10_gripper_controller/MoveRobot', MoveRobot)
        
        # Create an instance of the Pose message
        pose = Pose()
        
        # Create a request to the service
        response = move_robot(pose)
        
        # Check the result and print appropriate message
        if response.result:
            rospy.loginfo("Result: %s", response.result)
        else:
            rospy.logwarn("Service call failed")
    
    except rospy.ServiceException as e:
        rospy.logerr("Service call failed: %s", e)

if __name__ == "__main__":
    move_robot_client()